domino_data/_feature_store/run.py

```diff
@@ -31,10 +31,19 @@
         required=False,
         help="the feature store git repo branch",
     )
+
+    parser.add_argument(
+        "-r",
+        "--retry",
+        metavar="MAX_LOCK_RETRY_COUNT",
+        type=int,
+        required=False,
+        help="the maximum feature store lock retry count",
+    )
```

Comment:
so this argument would be used by the job script right? does this mean the user can somehow set how many retries they'd want for feature store syncs? where would they set that?

Reply:
This script is not supposed to be called by the users but by our front end. That's why we make the package name with the starting "_" to indicate this is something private. 

When the front end starts a job to do the synchronization, it's supposing to call the script 
"python3 -m domino_data.feature_store.run_featurestore_sync -r <max-retries>"

This is an optional argument, if not provided, it falls to the default 60. Basically retrying for 1 minute.

Reply:
got it!